compiler/semexprs.nim

```diff
@@ -1739,6 +1751,14 @@ proc goodLineInfo(arg: PNode): TLineInfo =
   else:
     arg.info
 
+proc isSubscrible(c: PContext, n: PNode): bool =
+  let idt = getIdent(c.graph.cache, "_")
+  let sid = nextSymId(c.idgen)
+  let temp = newSym(skTemp, idt, sid, c.p.owner, n.info, c.p.owner.options)
+  let tempAsNode = newSymNode(temp)
+  let tempSub = semExpr(c, newTupleAccessRaw(tempAsNode, 0), {efNoError, efFromUnpack})
+  return tempSub.kind != nkEmpty
```

Comment:
What?

Reply:
I don't understand what's the question

Reply:
this part maybe goes into lowerTupleUnpackingForAsgn I think

Reply:
You must not use `semExpr` just in order to ask for some related property. `semExpr` is full of side effects, it could run `echo` at compile-time.

Reply:
`echo` from user code or compiler code ? I haven't found other solution fit this case.

Reply:
Hmm...